### PR TITLE
[docs] Hello, Dagster becomes Quickstart

### PR DESCRIPTION
**Pull request by cmpadden**
## Summary & Motivation

We would like to reduce the barrier to entry by replacing Hello, Dagster with a Quickstart project. This project leverages GitHub Codespaces, so that users can try Dagster without installing any dependencies on their local machine.

This is a part of a larger initiative of simplifying the entrypoints into Dagster and our documentation.

### Checklist

- [ ] Make the https://github.com/dagster-io/dagster-quickstart repo public

### Outstanding Questions

- We have a few _quickstart_ templates that can already be scaffolded with the `dagster` command. Where do we want to draw the line on this quickstart vs those?

## How I Tested These Changes

- Ran Next.js project locally

### CODE REVIEW

**Comment by cmpadden** [comment]
Current dependencies on/for this PR:
* **#19981** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19981?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19981?utm_source=stack-comment).

File: docs/content/getting-started/quickstart.mdx [hunk 1]
```diff
@@ -0,0 +1,118 @@
+---
+title: Dagster Quickstart | Dagster Docs
+description: Run dagster for the first time
+---
+
+# Dagster Quickstart
+
+Welcome to Dagster! In this guide, you'll get up-and-running with the [Dagster Quickstart](https://github.com/dagster-io/dagster-quickstart) project. This project demonstrates the power of Dagster, and provides a starting point to allow further exploration of Dagster's features.
```
Comment (cmpadden) [inline docs/content/getting-started/quickstart.mdx:8]: Should we add a **NOTE** saying, "If you are looking to scaffold a new project instead, see ..."

Comment (cmpadden) [reply docs/content/getting-started/quickstart.mdx:8]: I've added it for now.

**Comment by cmpadden** [comment]
hyperlint review

**Review by PedramNavid — APPROVED** [review]
Just tested both the codespaces and local instructions and they work. I'll review the actual prose this afternoon, nice work!

File: docs/content/getting-started/quickstart.mdx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+5. **Success!**
+
+</TabItem>
+<TabItem name="Option 2: Running Locally">
+
+### Option 2: Running Locally
+
+<DagsterVersion />
+
+Ensure you have one of the supported Python versions installed before proceeding.
```
Comment (PedramNavid) [inline docs/content/getting-started/quickstart.mdx:58]: Maybe worth linking to an installation page?

**Comment by bllchmbrs** [comment]
@cmpadden - glad you're kicking the tires!

I've got some updates based on the team's feedback in #19175 and this one. I'd love to chat if you get the chance to hear your POV! Ping me here, linkedin, twitter, or through hyperlint support.

**Comment by cmpadden** [comment]
> @cmpadden - glad you're kicking the tires!
> 
> I've got some updates based on the team's feedback in #19175 and this one. I'd love to chat if you get the chance to hear your POV! Ping me here, linkedin, twitter, or through hyperlint support.

Thanks @bllchmbrs - apologies if I ran it on this PR a bit too early. It was funny being "called out" by Hyperlint for how passive I write. Overall, the suggestions were great, and I believe it's really improving the writing of our guides. Some observations:

- It created _a lot_ of recommendations, including a couple of files where I only made small tweaks. I suppose this was because it was the first time running the tool.
- It removed some personality in the writing, especially in some areas where I wanted to sound fun. I was able to edit the suggestion, and then merge it, which was handy.
- the Hyperlint summary for the PR seemed a bit repetitive
- The readability assessment was very insightful - that's a cool feature!

Overall I think it's super helpful, thanks again!

File: docs/content/getting-started/quickstart.mdx (hunk 2 above)
Comment (cmpadden) [reply docs/content/getting-started/quickstart.mdx:58]: Added a link, thanks.

**Comment by bllchmbrs** [comment]
@cmpadden thanks for the feedback. Every document has its own imperfections and couple that with the first time running it on`.mdx` and hyperlint got a little overzealous. Hyperlint gave an inordinate amount of feedback.

With that being said, I have already changed the PR review structure. GIve it a try on another PR and see what you think. It's much more concise and focuses on the key issues.

**Comment by cmpadden** [comment]
Hey @erinkcochran87 - this one is ready for your 👀 when you get the chance. Thanks!

File: docs/content/getting-started/quickstart.mdx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+1. Fork the [Dagster Quickstart](https://github.com/dagster-io/dagster-quickstart) repository
+
+2. Select **Create codespace on main** from the **Code** dropdown menu.
+
+<Image
+width={400}
+height={400}
+alt="Create codespace"
+src="/images/getting-started/quickstart/github-codespace-create.png"
+/>
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:67]: [2] This image isn't nested under the list item about it (`2`). Indent everything (I think it's 3 spaces; if not that, then 4) and everything should be in line

File: docs/content/getting-started/quickstart.mdx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+   ```
+
+4. Click **Open in Browser** when prompted.
+
+<Image
+width={400}
+height={300}
+alt="Codespace Open In Browser"
+src="/images/getting-started/quickstart/github-codespace-open-in-browser.png"
+/>
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:82]: [2]

File: docs/content/getting-started/quickstart.mdx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+5. **Success!**
+
+</TabItem>
+</TabGroup>
+
+## Navigating the User Interface
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:89]:
```suggestion
---

## Navigating the Dagster UI
```

[3] We use horizontal rules before every H2.

Also making this heading more specific by including the proper name of the UI

File: docs/content/getting-started/quickstart.mdx [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+<Image
+alt="HackerNews asset graph"
+src="/images/getting-started/quickstart/quickstart.png"
+width={2000}
+height={1956}
+/>
+
+But wait - there's more. Because the `hackernews_top_stories` asset returned some `metadata`, you can view the metadata right in the UI:
+
+1. Click the asset
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:113]:
```suggestion
1. Click the `hackernews_top_stories` asset.
```

Adding a period and including the name of the asset.

File: docs/content/getting-started/quickstart.mdx [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+2. In the sidebar, click the **Show Markdown** link in the **Materialization in Last Run** section. This opens a preview of the pipeline result, allowing you to view the top 10 HackerNews stories:
+
+<Image
+alt="Markdown preview of HackerNews top 10 stories"
+src="/images/getting-started/quickstart/hn-preview.png"
+width={2000}
+height={1754}
+/>
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:121]: [2]

File: docs/content/getting-started/quickstart.mdx [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+## Understanding the Code
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:123]:
```suggestion
---

## Understanding the code
```

[3] and using sentence casing

File: docs/content/getting-started/quickstart.mdx [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+
+The quickstart project defines two **Assets** using the `@asset` decorator:
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:125]: This is `Quickstart project` elsewhere in the guide. We should be consistent about how we write it.

Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:125]:
```suggestion
The quickstart project defines two **Assets** using the <PyObject object="asset" decorator /> decorator:
```

File: docs/content/getting-started/quickstart.mdx [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+
+    return MaterializeResult(
+        metadata={
+            "num_records": len(df),
+            "preview": MetadataValue.md(str(df[["title", "by", "url"]].to_markdown())),
+        }
+    )
+```
+
+The `hackernews_top_stories` asset returns a `MaterializeResult` with metadata, which is displayed in the Dagster UI, providing a markdown preview of the top stories.
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:180]: This feels like it's in a weird place - thoughts on combining it with the other sentence about this asset that comes before the code?

File: docs/content/getting-started/quickstart.mdx [hunk 11]
```diff
@@ -0,0 +1,198 @@
+---
+title: Dagster Quickstart | Dagster Docs
+description: Run dagster for the first time
+---
+
+# Getting Started with Dagster
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:6]:
```suggestion
# Quickstart
```

This is more descriptive of the guide as a whole.

File: docs/content/getting-started/quickstart.mdx [hunk 12]
```diff
@@ -0,0 +1,198 @@
+---
+title: Dagster Quickstart | Dagster Docs
```
Comment (erinkcochran87) [inline docs/content/getting-started/quickstart.mdx:2]:
```suggestion
title: Quickstart | Dagster Docs
```

File: docs/content/getting-started/quickstart.mdx [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+
+# Getting Started with Dagster
+
+<Note>
+  Looking to scaffold a new project? Check out the{" "}
+  <Link href="/getting-started/create-new-project">Creating a New Project</Link>{" "}
+  guide!
+</Note>
+
+## Introduction
```
[the remark this answers is not in the document]
Comment (cmpadden) [reply docs/content/getting-started/quickstart.mdx:14]:
```suggest
  guide!
</Note>
```

File: docs/content/getting-started/quickstart.mdx (hunk 10 above)
Comment (cmpadden) [reply docs/content/getting-started/quickstart.mdx]: Good suggestion, thanks.

**Pull request merged**